--- openttd-1.2.2/config.lib~	2012-09-24 09:09:54.018404061 +0200
+++ openttd-1.2.2/config.lib	2012-09-24 09:17:48.355053988 +0200
@@ -2831,7 +2831,7 @@
 
 	version=`$fontconfig_config --modversion 2>/dev/null`
 	ret=$?
-	shortversion=`echo $version | cut -c 1,3`
+	shortversion=`echo $version | cut -d . -f 1,2 --output-delimiter=`
 	log 2 "executing $fontconfig_config --modversion"
 	log 2 "  returned $version"
 	log 2 "  exit code $ret"
