--- openttd-1.2.2/config.lib~	2012-09-24 09:09:54.018404061 +0200
+++ openttd-1.2.2/config.lib	2012-09-24 09:17:48.355053988 +0200
@@ -2831,7 +2831,7 @@
 		fontconfig_config="$with_fontconfig"
 	fi
 
-	version=`$fontconfig_config --modversion 2>/dev/null`
+ 	version=`$fontconfig_config --modversion 2>/dev/null | cut -d . -f 1,2 --output-delimiter=`
 	ret=$?
 	check_version '2.3' "$version"
 	version_ok=$?
