diff -dur -x '*.orig' -x '*.rej' -x '*~' openttd-1.4.0.orig/config.lib openttd-1.4.0/config.lib
--- openttd-1.4.0.orig/config.lib	2014-04-01 20:37:07.000000000 +0200
+++ openttd-1.4.0/config.lib	2014-04-08 19:28:02.013554008 +0200
@@ -69,6 +69,7 @@
 	with_allegro="1"
 	with_sdl="1"
 	with_cocoa="1"
+	with_home_etc="1"
 	with_zlib="1"
 	with_lzma="1"
 	with_lzo2="1"
@@ -144,6 +145,7 @@
 		with_allegro
 		with_sdl
 		with_cocoa
+		with_home_etc
 		with_zlib
 		with_lzma
 		with_lzo2
@@ -333,6 +335,9 @@
 			--without-cocoa)              with_cocoa="0";;
 			--with-cocoa=*)               with_cocoa="$optarg";;
 
+			--with-home-etc)              with_home_etc="1";;
+			--without-home-etc)           with_home_etc="0";;
+
 			--with-zlib)                  with_zlib="2";;
 			--without-zlib)               with_zlib="0";;
 			--with-zlib=*)                with_zlib="$optarg";;
@@ -1651,6 +1656,11 @@
 		fi
 	fi
 
+	if [ "$with_home_etc" != 0 ]; then
+		CFLAGS="$CFLAGS -DWITH_HOME_ETC"
+		LIBS="$LIBS -lhome_etc"
+	fi
+
 	if [ "$with_zlib" != "0" ]; then
 		if [ "$enable_static" != "0" ] && [ "$os" != "OSX" ]; then
 			LIBS="$LIBS $zlib"
diff -dur -x '*.orig' -x '*.rej' -x '*~' openttd-1.4.0.orig/src/fileio.cpp openttd-1.4.0/src/fileio.cpp
--- openttd-1.4.0.orig/src/fileio.cpp	2014-04-01 20:37:03.000000000 +0200
+++ openttd-1.4.0/src/fileio.cpp	2014-04-08 19:30:30.296888687 +0200
@@ -32,6 +32,10 @@
 #include "basedir.h"
 #endif
 
+#ifdef HOME_ETC
+#include <home_etc.h>
+#endif
+
 /** Size of the #Fio data buffer. */
 #define FIO_BUFFER_SIZE 512
 
@@ -1103,7 +1107,11 @@
 	 * or at least before something else touches the environment
 	 * variables in any way. It can also contain all kinds of
 	 * unvalidated data we rather not want internally. */
+#ifdef HOME_ETC
+	const char *homedir = _HEdir;
+#else
 	const char *homedir = getenv("HOME");
+#endif
 	if (homedir != NULL) {
 		homedir = strndup(homedir, MAX_PATH);
 	}
