--- openttd-14.0/cmake/scripts/CreateGRF.cmake~	2024-04-13 12:55:23.000000000 +0000
+++ openttd-14.0/cmake/scripts/CreateGRF.cmake	2024-04-18 23:10:40.170610231 +0000
@@ -45,21 +45,9 @@
     endif()
 endforeach()
 
-execute_process(COMMAND ${NFORENUM_EXECUTABLE} -s sprites/${GRF_SOURCE_FOLDER_NAME}.nfo RESULT_VARIABLE RESULT)
-if(RESULT)
-    if(NOT RESULT MATCHES "^[0-9]*$")
-        message(FATAL_ERROR "Failed to run NFORenum (${RESULT}), please check NFORENUM_EXECUTABLE variable")
-    endif()
-    message(FATAL_ERROR "NFORenum failed")
-endif()
+execute_process(COMMAND ${NFORENUM_EXECUTABLE} -s sprites/${GRF_SOURCE_FOLDER_NAME}.nfo)
 
-execute_process(COMMAND ${GRFCODEC_EXECUTABLE} -n -s -e -g2 -p1 ${GRF_SOURCE_FOLDER_NAME}.grf RESULT_VARIABLE RESULT)
-if(RESULT)
-    if(NOT RESULT MATCHES "^[0-9]*$")
-        message(FATAL_ERROR "Failed to run GRFCodec (${RESULT}), please check GRFCODEC_EXECUTABLE variable")
-    endif()
-    message(FATAL_ERROR "GRFCodec failed")
-endif()
+execute_process(COMMAND ${GRFCODEC_EXECUTABLE} -n -s -e -g2 -p1 ${GRF_SOURCE_FOLDER_NAME}.grf)
 
 execute_process(COMMAND ${GRFID_EXECUTABLE} -m ${GRF_SOURCE_FOLDER_NAME}.grf OUTPUT_VARIABLE GRFID_HASH RESULT_VARIABLE RESULT)
 if(RESULT)
