--- openttd/stdafx.h.orig	2004-06-21 19:23:15.000000000 +0000
+++ openttd/stdafx.h	2004-06-26 12:34:28.000000000 +0000
@@ -81,13 +81,24 @@
 
 // Check endianness
 #if !defined(TTD_LITTLE_ENDIAN) && !defined(TTD_BIG_ENDIAN)
+#  ifdef __linux__
+#	include <endian.h>
+#	if __BYTE_ORDER == __LITTLE_ENDIAN
+#		define TTD_LITTLE_ENDIAN
+#	elif __BYTE_ORDER == __BIG_ENDIAN
+#		define TTD_BIG_ENDIAN
+#	else
+#		error No endianness defined, use either TTD_LITTLE_ENDIAN or TTD_BIG_ENDIAN
+#	endif
+#  else
 #	if defined(_BIG_ENDIAN)
 #		define TTD_BIG_ENDIAN
 #	elif defined(_LITTLE_ENDIAN)
 #		define TTD_LITTLE_ENDIAN
 #	else
 #		error No endianness defined, use either TTD_LITTLE_ENDIAN or TTD_BIG_ENDIAN
-#endif
+#	endif
+#  endif
 #endif
 
 #if defined(UNIX)
